### ISSUE TITLE
[Checkbox][a11y] Unable to check checkbox with voiceover 

### ISSUE THREAD

**Issue by WillsonSmith** [Shopify/polaris-react#3973]
## Issue summary
When using voiceover on either iOS or macOS, the voiceover command for click (iOS: Double tap, macOS: VO + Space) does not toggle the checkbox.

## Expected behavior
When you click the checkbox with voiceover, it should select or deselect the checkbox.

## Actual behavior
When you click the checkbox with voiceover, it does not select the checkbox.

Recordings

iOS

https://user-images.githubusercontent.com/1087756/106942537-4aeb8580-66f2-11eb-81ed-c61753b7f308.mp4

macOS

https://user-images.githubusercontent.com/1087756/106942370-1bd51400-66f2-11eb-86c1-92ca45930ad7.mp4

## Steps to reproduce the problem

1. Navigate to a page with a checkbox (https://wlmtq.csb.app)
2. Turn on voiceover and tab to the checkbox
3. When prompted, use the voiceover command to toggle the checkbox

## Reduced test case
https://wlmtq.csb.app

## Specifications

- Are you using the React components? (Y/N): Y
- Polaris version number: Latest (& 5.15.1/any)
- Browser: Any
- Device: Macbook pro, iPhone
- Operating System: macOS 11, iOS 14.4
